### PR TITLE
Refactor Python scripts in build.rs

### PR DESCRIPTION
**Pull request by kngwyu**
I also considered placing these scripts in separate files and applying black, but it seemed overkill.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -690,30 +690,34 @@ fn find_interpreter_and_get_config() -> Result<(InterpreterConfig, BuildFlags)>
...
 print("version_major", sys.version_info[0])
 print("version_minor", sys.version_info[1])
 print("implementation", platform.python_implementation())
 if libdir is not None:
     print("libdir", libdir)
-print("ld_version", sysconfig.get_config_var('LDVERSION') or sysconfig.get_config_var('py_version_short'))
+print(
+    "ld_version",
+    ld_version or sysconfig.get_config_var("py_version_short"),
+)
```
Comment (kngwyu) [inline build.rs:717]: I formatted this by black with its default line width (=88).

Comment (davidhewitt) [reply build.rs:717]: Might be able to reduce line width for this and the lines below by caching `sysconfig.get_config_var`, e.g.:
```suggestion
var = sysconfig.get_config_var
print("ld_version", ld_version or var("py_version_short"))
```

**Review by davidhewitt — APPROVED** [review]
👍 thanks for cleaning these up! Just one optional suggestion.

File: build.rs (hunk 1 above)
Comment (kngwyu) [reply build.rs:717]: Thanks, I choose `from sysconfig import get_config_var`.

**Comment by davidhewitt** [comment]
Even better! Thanks 🚀

**Pull request merged**